libs/core/src/lib/select/commons/select-commons.ts

```diff
@@ -0,0 +1,994 @@
+/** The MIT License
+* Copyright (c) 2014-2016 Google, Inc. http://angularjs.org
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy of this software and 
+* associated documentation files (the "Software"), to deal in the Software without restriction,
+* including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+* and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
+* subject to the following conditions:
+* The above copyright notice and this permission notice shall be
+* included in all copies or substantial portions of the Software.
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+* INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE
+* AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+* DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*
+*
+* */
+import { ActiveDescendantKeyManager, LiveAnnouncer } from '@angular/cdk/a11y';
+import { FdOptionSelectionChange, OptionComponent } from '../option/option.component';
+import { FdSelectChange } from '../select.component';
+import { ChangeDetectorRef, ElementRef, EventEmitter, isDevMode, NgZone, QueryList, Input } from '@angular/core';
+import { Directionality } from '@angular/cdk/bidi';
+import { defer, merge, Observable, Subject } from 'rxjs';
+import { CdkConnectedOverlay, ConnectedPosition, ViewportRuler } from '@angular/cdk/overlay';
+import { SelectionModel } from '@angular/cdk/collections';
+import { NgControl } from '@angular/forms';
+import { startWith, switchMap, take, takeUntil } from 'rxjs/operators';
+import { DOWN_ARROW, END, ENTER, hasModifierKey, HOME, SPACE, UP_ARROW } from '@angular/cdk/keycodes';
+
+
+ import { KeyUtil } from '../../utils/functions/key-util';
+
+/**
+ * Common behaviors from Material Select Component to build Fundamental-ngx Select on top of it. We need this solid
+ * foundation to deliver robust solution and only focus on additional Fiori 3 enhancement.
+ *
+ * Even this seems to be implemented as one big mixins with single anonymous class all this is going to be broking down
+ * into several pieces in order to maximize reusability. Such as a possible candidates would be to separate
+ * Keyboard Support, Overlay support and option panel positioning and maybe select behavior itself.
+ *
+ * @Hidden
+ */
```

Comment:
?

Reply:
many of the method that are private needs to be public. We need to also update the API docs. In development you always have cases where: 

private _xxxx -> you dont want anybody to use it from outside
_xxxxx -> Private API , but other classes access it internaly from outside (it can change)